consensus/types/src/data_column_subnet_id.rs

```diff
@@ -66,17 +69,37 @@ impl DataColumnSubnetId {
             }
             current_id += U256::one()
         }
-        subnets.into_iter().map(DataColumnSubnetId::new)
+        Ok(subnets.into_iter().map(DataColumnSubnetId::new))
+    }
+
+    /// Compute the custody subnets for a given node id with the default `custody_requirement`.
+    /// This operation should be infallable, and empty iterator is returned if it fails unexpectedly.
+    pub fn compute_custody_requirement_subnets<E: EthSpec>(
+        node_id: U256,
+        spec: &ChainSpec,
+        log: &Logger,
+    ) -> impl Iterator<Item = DataColumnSubnetId> {
+        Self::compute_custody_subnets::<E>(node_id, spec.custody_requirement, spec)
+            .map(Either::Left)
+            .unwrap_or_else(|e| {
+                error!(
+                    log,
+                    "Failed to compute default custody subnets for node";
+                    "node_id" => %node_id,
+                    "info" => "This is a bug (unreachable case), please notify the devs",
+                    "error" => ?e
+                );
+                Either::Right(std::iter::empty())
```

Comment:
Why not panic here? You can add a unit test ensuring that `custody_requirement < data_column_sidecar_subnet_count` in all our known specs. If someone tries to run a network with broken config the panic should be ok.

Reply:
We probably could. My thinking was because `compute_custody_subnets` return `Error` type in the error case, it could return other variants from the API perspective - although the current implementation could only fail due to `InvalidCustodySubnetCount`, which _could_ change. I usually prefer to avoid assumptions on implementation but in this case this is an implementation on the same struct, so I guess it's fine.

Reply:
I'm slightly concerned if this is somehow not caught on startup - e.g. all devnet peers provide their `csc` override and we never use the default. I'll add an unit test to this function.

Reply:
Added test here. https://github.com/sigp/lighthouse/pull/6308/commits/763ea039b04a5b47918adcd28634edae9165392c